### PR TITLE
add(rpc): getblock: return transaction details with verbosity=2

### LINKED ISSUE #9024: getblock: return transactions details with `verbosity=2`
### Motivation

For zcashd compatibility. In verbose=2 mode, the transactions details should be returned instead of transaction hashes.

### Specifications

https://github.com/zcash/zcash/blob/99ad6fdc3a549ab510422820eea5e5ce9f60a5fd/src/rpc/blockchain.cpp#L797-L799

### Complex Code or Requirements

Can probably be done by calling getrawtransaction to reuse its logic.

### Testing

_No response_

### Related Work

_No response_

### PR DESCRIPTION
**Pull request by conradoplg**
## Motivation

We want getblock to match zcashd. With verbosity=2 we returned a list of hashes, but a list of transaction objects is expected.

Closes #9024 

### Specifications & References

https://github.com/zcash/zcash/blob/99ad6fdc3a549ab510422820eea5e5ce9f60a5fd/src/rpc/blockchain.cpp#L735

## Solution

If `verbosity=2`, call `getrawtransaction` for each tx in the block internally and use its output to fill the transaction details.

There are tons of fields missing in `getrawtransaction`, but that will be another issue. When they are filled, they will be automatically returned by `getblock` too.

`zcashd` seems to not return the `heigth` and `confirmations` fields in the transactions details inside `getblock` (which are indeed redundant) but I think it's harmless to return them.

### Tests

Expanded existing test. Also tested manually with `zcash-cli`.

### Follow-up Work

See https://github.com/ZcashFoundation/zebra/issues/8446 for other `getblock` related stuff, and we'll need an issue for completing `getrawtransaction`

### PR Author's Checklist

- [ ] The PR name will make sense to users.
- [ ] The PR provides a CHANGELOG summary.
- [ ] The solution is tested.
- [ ] The documentation is up to date.
- [ ] The PR has a priority label.

### PR Reviewer's Checklist

- [ ] The PR Author's checklist is complete.
- [ ] The PR resolves the issue.

### CODE REVIEW

File: zebra-rpc/src/methods.rs [hunk 1]
```diff
@@ -1987,22 +2021,33 @@ pub enum GetRawTransaction {
...
 }
 
 impl Default for GetRawTransaction {
     fn default() -> Self {
-        Self::Object {
+        Self::Object(TransactionObject::default())
+    }
+}
+
+/// A Transaction object as returnedb by `getrawtransaction` RPC request.
```
Comment (oxarbitrage) [inline zebra-rpc/src/methods.rs]:
```suggestion
/// A Transaction object as returned by `getrawtransaction` RPC request.
```

**Review by oxarbitrage — DISMISSED** [review]
Looks good, it might conflict with https://github.com/ZcashFoundation/zebra/pull/9049

**Comment by oxarbitrage** [comment]
It seems some test is still locked somewhere. I am unsure if it is caused by this code but it seems so.

**Comment by arya2** [comment]
This looks good, I opened a suggestion PR (https://github.com/ZcashFoundation/zebra/pull/9084) for reducing db queries and service calls.

**Review by oxarbitrage — APPROVED** [review]

**Pull request merged**